Python Azure: Add installation instructions for Azure in Python (#18867)
newrelic/docs-website: src/content/docs/apm/agents/python-agent/hosting-services/python-azure-containerapps-appservice.mdx

## Install the Python agent onto Azure App Service or Container Apps[#install-python]

To add the New Relic Python agent to your Azure application, add `newrelic` to your `requirements.txt` file.

Here's an example with a sample app:

```python
# app.py
# Run with command "flask run"

from flask import Flask

app = Flask(__name__)

@app.route('/')
def hello_world():
    return "Hello, fellow humans!"

if __name__ == '__main__':
    app.run()
```

On the command line:

```console
foo@bar:~$ cat requirements.txt 
newrelic
flask
```

If you're using a Dockerfile, add `newrelic-admin run-program` as an entrypoint to your startup script.

```lang-dockerfile
COPY requirements.txt .
COPY app.py .

RUN pip install -r requirements.txt

EXPOSE 5000

ENTRYPOINT ["newrelic-admin", "run-program"]
CMD ["flask", "run"]
```

Otherwise, initialize the agent by adding these lines to your app:

```python
# app.py
# Run with command "flask run"

> import newrelic.agent   # ADDED

Reviewer: Is initializing the same as adding the Python agent?

Author: Sort of--the way the agent works is that you install the agent on the host (in this case, Azure), but you have to do something to let the app know that it needs to run the code with New Relic.  This is done either with the New Relic commands in front of the startup script or the initialization function.

TL;DR: I wanted to distinguish between installing New Relic and actually flagging New Relic to run on the app

Reviewer: Ah ok, and you install New Relic Python agent to the Azure app by running a bash command, "adding" New Relic to the flask app

    ```

    </Collapser>
    <Collapser
        id="add-via-docker"
        title="Add the Python agent to a Dockerfile"
    >
    
    If you're using a Dockerfile, add `newrelic-admin run-program` as an entrypoint to your startup script.

    ```go
        COPY requirements.txt .
        COPY app.py .

        RUN pip install -r requirements.txt

        EXPOSE 5000

        ENTRYPOINT ["newrelic-admin", "run-program"]
        CMD ["flask", "run"]
    ```

    </Collapser>
</CollapserGroup>

    </Step>
    <Step>

    ## Initialize the agent on app start-up [#initialize-agent]

Update your app code to import `newrelic.agent`, then add the `newrelic.agent.initialize()` function to initialize the agent. 

Here's an example Flask app with the agent initialized:

    ```python
    # app.py
    # Run with command "flask run"

    import newrelic.agent   # ADDED
    from flask import Flask

    newrelic.agent.initialize()   # ADDED

    app = Flask(__name__)

    @app.route('/')
    def hello_world():
        return "Hello, fellow humans!"

    if __name__ == '__main__':
        app.run()
    ```
    </Step>
    <Step>
        ## Configure your Azure settings [#azure-settings]

> Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python SDK CLI. 

Author: In this case, it's just the "Python CLI" instead of the "Python SDK CLI"

> Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python SDK CLI. 

Reviewer: ```suggestion
Once you've installed the Python agent, we recommend configuring your app in either the Azure App Service or Container Apps portal, depending on the service you're using. You can update your app settings through the Azure UI, or using the Python CLI. 
```

<CollapserGroup>
    <Collapser
        id="config-app-services-ui"

>         title="Configure your app in Azure App Services"

Author: ```suggestion
        title="Configure your app in Azure App Services through the Azure Portal"
```

    >

1. Sign in to <DNT>**[portal.azure.com](https://portal.azure.com)**</DNT>.
2. Select <DNT>**App Services > (select a Python app) > Settings > Environment variables**</DNT>.
3. Add the following to the <DNT>**App settings**</DNT>:
    * `NEW_RELIC_APP_NAME`: Your Linux Azure website name
    * `NEW_RELIC_LICENSE_KEY`: Your New Relic <InlinePopover type="licenseKey" />
4. Save your settings.
5. Restart your Python app.

    </Collapser>
    <Collapser
        id="config-container-apps-ui"

>         title="Configure your app in Azure Container Apps"

Author: ```suggestion
        title="Configure your app in Azure Container Apps though the Azure Portal"
```

    >
1. Sign in to <DNT>**[portal.azure.com](https://portal.azure.com)**</DNT>.
2. Select <DNT>**Container Apps > (select a Python app) > Containers > Environment variables**</DNT>.
3. Add the following to the <DNT>**Container settings**</DNT>:

   * `NEW_RELIC_APP_NAME`: Your Linux Azure website name
   * `NEW_RELIC_LICENSE_KEY`: Your New Relic <InlinePopover type="licenseKey" />
4. Save your settings.
5. Restart your Python app.

Generate some traffic to your app and wait a few minutes. You can see your data by going to <DNT>**[one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > APM & services > (select your app)**</DNT>.
    </Collapser>
    <Collapser
    id="config-app-services-cli"

>     title="Configure your app in Azure Container Apps"

Author: ```suggestion
    title="Configure your app in Azure App Services through the CLI"
```

    >

Run the following on the command line: 

```bash
az webapp config appsettings set --name ${WEB_APP_NAME} --settings NEW_RELIC_LICENSE_KEY=${NEW_RELIC_LICENSE_KEY} NEW_RELIC_APP_NAME=${NEW_RELIC_APP_NAME}
```

    </Collapser>
    <Collapser
    id="config-container-apps-cli"

>     title="Configure your app in Azure Container Apps"

Author: ```suggestion
    title="Configure your app in Azure Container Apps though the CLI"
```
